src/RamExecutor.cpp

```diff
@@ -2567,12 +2680,14 @@ std::string RamCompiler::compileToBinary(const SymbolTable& symTable, const RamP
     }
 
     // done
-    return binary;
+    return source;
```

Comment:
Are you returning the filename of the binary produced or of the source code file? (The naming in this method was distinctly unclear well before you touched it.)